addons/sale_project/models/sale_order.py

```diff
@@ -12,7 +12,7 @@
 class SaleOrder(models.Model):
     _inherit = 'sale.order'
 
-    tasks_ids = fields.Many2many('project.task', compute='_compute_tasks_ids', search='_search_tasks_ids', string='Tasks associated with this sale', export_string_translation=False)
+    tasks_ids = fields.Many2many('project.task', compute='_compute_tasks_ids', search='_search_tasks_ids', groups="project.group_project_user", string='Tasks associated with this sale', export_string_translation=False)
     tasks_count = fields.Integer(string='Tasks', compute='_compute_tasks_ids', groups="project.group_project_user", export_string_translation=False)
```

Comment:
I disagree with the motivation for aligning groups on fields with the same compute method, although in this case the group restrictions are nonsensical for another reason.

Groups on fields should determine whether you can read them, and how you determine the value of those fields should be irrelevant.  It makes sense to be able to see how many tasks without being able to access those tasks.  The groups/accesses you need to compute those values is a different issue: you may restrict the access to the fields, but have the computation done in sudo mode, for instance.

Reply:
Ok, but you *should* have two different functions computing the values if you have different access rights, as using sudo bypasses access rights and could change the number of tasks or task ids.

Anyways, I will reformulate the commit message.

Reply:
You are talking about something else: the compute method in that case is inconsistent.  The result should probably not depend on the user or the environment with which you access/compute the field.  If that's the intention, the field should depend on context key `'uid'` (which is a shortcut for `(env.uid, env.su)`).

Reply:
You may turn the argument upside down: if you can see the tasks, you should see the count.
Simply ask the responsible team what the policy should be.

By the way, the fact that you need read access on `tasks_count` is a consequence of badly written code.  You may even `sudo()` the record for that, if it fixes the problem.